src/LuaTransformer.ts

```diff
@@ -4699,6 +4691,15 @@ export class LuaTransformer {
             return this.createExportedIdentifier(identifier, exportScope);
         }
 
+        if (expression.originalKeywordKind === ts.SyntaxKind.UndefinedKeyword) {
+            return tstl.createNilLiteral();
+        }
+
+        const identifierType = this.checker.getTypeAtLocation(expression);
+        if (identifierType.symbol && identifierType.symbol.escapedName === "globalThis") {
```

Comment:
I think we shouldn't use type there and only transform actual `globalThis` identifier.
Declaring a variable is just a way to tell type system about already existing thing, so transforming it to something else would be really unexpected behavior for me.
Also not transforming it would avoid confusing things like an object property with type of globalThis working only when it's assigned to a variable.

Reply:
The idea was to allow the extension `typeof globalThis` e.g:
```
interface Test {
    foo: string;
}
declare let globalAlias: typeof globalThis & Test;
globalAlias.foo = "bar";
```
This would allow typesafe access to `_G`. Without  the ability to extend `typeof globalThis` you will only be able to access global variables that are declared in Lua by casting to any `(print(<any>globalThis.myLuaGlobal))`,

But you are right this would conflict with stuff like object properties that are assigned to globalThis.
I reverted the changes in the latest commit.